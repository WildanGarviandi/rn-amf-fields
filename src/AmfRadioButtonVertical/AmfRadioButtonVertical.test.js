import React from 'react'
import AmfRadioButtonVertical from './AmfRadioButtonVertical'
import renderer from 'react-test-renderer'

const label = 'Label'
const items = [{
  label: 'foo',
  value: 'foo'
}, {
  label: 'bar',
  value: 'bar'
}]
const onPress = jest.fn()

test('AmfRadioButtonVertical simple render', () => {
  let props = { label, items, onPress }
  props.options = [
    { label: 'Label1', value: 'value2' },
    { label: 'Label2', value: 'value3' },
  ]

  const component = renderer.create(<AmfRadioButtonVertical {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test.skip('AmfRadioButtonVertical with validation', () => {
  let props = { label, items, onPress,
    required: true,
    value: ''
  }

  const component = renderer.create(<AmfRadioButtonVertical {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

/**
 *
 */
test('AmfRadioButtonVertical without options', () => {
  let props = { label, items, onPress }

  console.error = jest.fn()

  expect(() => {
    const component = renderer.create(<AmfRadioButtonVertical {...props} />)
    let tree = component.toJSON()
  }).toThrowErrorMatchingSnapshot()
})
