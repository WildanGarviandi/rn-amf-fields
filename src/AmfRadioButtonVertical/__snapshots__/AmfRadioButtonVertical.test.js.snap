// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfRadioButtonVertical simple render 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
        "marginRight": 16,
      }
    }
  >
    Label
  </Text>
  <View
    style={
      Array [
        undefined,
        Object {
          "alignItems": "flex-start",
        },
        false,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#000000",
                "borderRadius": 12,
                "borderWidth": 1,
                "height": 24,
                "justifyContent": "center",
                "opacity": 1,
                "width": 24,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "flexWrap": "wrap",
              },
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "lineHeight": 20,
                  "paddingLeft": 10,
                },
                false,
                Object {
                  "color": undefined,
                },
                Object {},
              ]
            }
          >
            Label1
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#000000",
                "borderRadius": 12,
                "borderWidth": 1,
                "height": 24,
                "justifyContent": "center",
                "opacity": 1,
                "width": 24,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "flexWrap": "wrap",
              },
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "lineHeight": 20,
                  "paddingLeft": 10,
                },
                false,
                Object {
                  "color": undefined,
                },
                Object {},
              ]
            }
          >
            Label2
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`AmfRadioButtonVertical without options 1`] = `"Field \\"Label\\" (AmfRadioButton) doesnt have options"`;
